src/utils/TypeGuards.ts

```diff
@@ -857,7 +858,9 @@ export class TypeGuards {
     /**
      * Gets if the node is a InferKeyword.
      */
-    static readonly isInferKeyword: (node: compiler.Node) => node is compiler.Node = TypeGuards.is(SyntaxKind.InferKeyword);
+    static isInferKeyword(node: compiler.Node): node is compiler.Node<Token<SyntaxKind.InferKeyword>> { 
+        return node.getKind() === SyntaxKind.InferKeyword;
+    }

```

Comment:
This file is actually code generated so we can't make this change here. For example, this PR did the change to make everything in here use `TypeGuards.is`: https://github.com/dsherret/ts-morph/pull/703/files

Do you want to merge the jsdoc change above separately as that's a good change? Then we could come up for a fix for this separately. Let me know if you'd like to look at fixing the generated code or I can take a look at it soon.

Reply:
I saw the note in the header comment but thought only applied to the actual method is()

I will fix there then. 

do you agree with that type casting ?

Reply:
Oh yeah, I can see how that header comment might be confusing. Feel free to fix it!

Do you mean casting the result of `TypeGuards.is(SyntaxKind.InferKeyword)` to `compiler.Node<Token<SyntaxKind.InferKeyword>>`? That should be fine!

Thanks again! 😊